### PR TITLE
fix: disable submission in template mode

### PR DESCRIPTION
**Pull request by wanlingt**
## Problem

Forms do not need to be submitted in template mode - refer to [this comment](https://github.com/opengovsg/FormSG/pull/5377#discussion_r1028659573) for context

## Solution

Continuation of #5377 

Disable `PublicFormSubmitButton` if the form is in template mode

## Before & After Screenshots

**AFTER**:
image

### CODE REVIEW

File: frontend/src/features/admin-form/template/TemplateFormProvider.tsx [hunk 1]
```diff
@@ -101,63 +101,14 @@ export const TemplateFormProvider = ({
     [data?.form, data?.spcpSession],
   )
 
-  const { submitEmailModeFormMutation, submitStorageModeFormMutation } =
-    usePreviewFormMutations(formId)
-
+  // Forms are not submitted in template mode,
+  // hence this SubmitHandler does not contain submitEmailModeFormMutation and submitStorageModeFormMutation
   const handleSubmitForm: SubmitHandler<FormFieldValues> = useCallback(
```
Comment (karrui) [inline frontend/src/features/admin-form/template/TemplateFormProvider.tsx]: We may not need this function at all right? 

Should we update the context return type to be either a function or explicitly undefined (so devs will not accidentally forget the function), and use the presence of the function as a source of truth of whether the submit button should be disabled or not? 

This way we do not need a new variable

File: frontend/src/features/public-form/PublicFormContext.tsx [hunk 2]
```diff
@@ -45,6 +45,7 @@ export interface PublicFormContextProps
    */
   onMobileDrawerOpen: () => void
   onMobileDrawerClose: () => void
+  isTemplate?: boolean
```
Comment (karrui) [inline frontend/src/features/public-form/PublicFormContext.tsx]: If we use the presence of a submit function we do not need this prop at all

File: frontend/src/features/public-form/components/FormFields/PublicFormSubmitButton.tsx [hunk 3]
```diff
@@ -49,7 +51,7 @@ export const PublicFormSubmitButton = ({
         colorScheme={`theme-${colorTheme}` as ThemeColorScheme}
         type="button"
         isLoading={isSubmitting}
-        isDisabled={!!preventSubmissionLogic}
+        isDisabled={!!preventSubmissionLogic || !!isTemplate}
```
Comment (karrui) [inline frontend/src/features/public-form/components/FormFields/PublicFormSubmitButton.tsx]:
```suggestion
        isDisabled={!!preventSubmissionLogic || !onSubmit}
```

File: frontend/src/features/admin-form/template/TemplateFormProvider.tsx (hunk 1 above)
Comment (wanlingt) [reply frontend/src/features/admin-form/template/TemplateFormProvider.tsx]: [656ff12](https://github.com/opengovsg/FormSG/pull/5443/commits/656ff12b7b65076e95c6cc1199bf856d50e05551)

File: frontend/src/features/admin-form/template/TemplateFormProvider.tsx [hunk 4]
```diff
@@ -101,73 +84,14 @@ export const TemplateFormProvider = ({
...
   }
 
   return (
     <PublicFormContext.Provider
       value={{
-        handleSubmitForm,
+        handleSubmitForm: undefined,
         formId,
         error,
         submissionData,
```
Comment (karrui) [inline frontend/src/features/admin-form/template/TemplateFormProvider.tsx]: technically this can be undefined too without the useState initialisation

**Review by karrui — APPROVED** [review]
lgtm

File: frontend/src/features/admin-form/template/TemplateFormProvider.tsx (hunk 4 above)
Comment (wanlingt) [reply frontend/src/features/admin-form/template/TemplateFormProvider.tsx]: [564ef08](https://github.com/opengovsg/FormSG/pull/5443/commits/564ef08fef4d4645e58581906547cfb923ad9c90)

**Review by karrui — APPROVED** [review]
lgtm

**Pull request merged**